packages/rocketchat-slackbridge/rocket.js

```diff
@@ -0,0 +1,421 @@
+
+			if (alias !== rocketUserName) {
+				rocketMsgObj.alias = alias;
+			}
+		}
+
+		return rocketMsgObj;
+	}
+
+	createAndSaveMessage(rocketChannel, rocketUser, slackMessage, rocketMsgDataDefaults, isImporting) {
+		if (slackMessage.type === 'message') {
+			let rocketMsgObj = {};
+			if (!_.isEmpty(slackMessage.subtype)) {
+				rocketMsgObj = this.slack.processSubtypedMessage(rocketChannel, rocketUser, slackMessage, isImporting);
+				if (!rocketMsgObj) {
+					return;
+				}
+			} else {
+				rocketMsgObj = {
+					msg: this.convertSlackMsgTxtToRocketTxtFormat(slackMessage.text),
+					rid: rocketChannel._id,
+					u: {
+						_id: rocketUser._id,
+						username: rocketUser.username
+					}
+				};
+
+				this.addAliasToMsg(rocketUser.username, rocketMsgObj);
+			}
+			_.extend(rocketMsgObj, rocketMsgDataDefaults);
+			if (slackMessage.edited) {
+				rocketMsgObj.editedAt = new Date(parseInt(slackMessage.edited.ts.split('.')[0]) * 1000);
+			}
+			if (slackMessage.subtype === 'bot_message') {
+				rocketUser = RocketChat.models.Users.findOneById('rocket.cat', { fields: { username: 1 } });
+			}
+
+			if (slackMessage.pinned_to && slackMessage.pinned_to.indexOf(slackMessage.channel) !== -1) {
+				rocketMsgObj.pinned = true;
+				rocketMsgObj.pinnedAt = Date.now;
+				rocketMsgObj.pinnedBy = _.pick(rocketUser, '_id', 'username');
+			}
+			if (slackMessage.subtype === 'bot_message') {
+				Meteor.setTimeout(() => {
+					if (slackMessage.bot_id && slackMessage.ts && !RocketChat.models.Messages.findOneBySlackBotIdAndSlackTs(slackMessage.bot_id, slackMessage.ts)) {
+						RocketChat.sendMessage(rocketUser, rocketMsgObj, rocketChannel, true);
+					}
+				}, 500);
+			} else {
+				logger.rocket.debug('Send message to Rocket.Chat');
+				RocketChat.sendMessage(rocketUser, rocketMsgObj, rocketChannel, true);
+			}
+		}
+	}
+
+	convertSlackMsgTxtToRocketTxtFormat(slackMsgTxt) {
+		if (!_.isEmpty(slackMsgTxt)) {
+			slackMsgTxt = slackMsgTxt.replace(/<!everyone>/g, '@all');
+			slackMsgTxt = slackMsgTxt.replace(/<!channel>/g, '@all');
+			slackMsgTxt = slackMsgTxt.replace(/&gt;/g, '<');
+			slackMsgTxt = slackMsgTxt.replace(/&lt;/g, '>');
+			slackMsgTxt = slackMsgTxt.replace(/&amp;/g, '&');
+			slackMsgTxt = slackMsgTxt.replace(/:simple_smile:/g, ':smile:');
+			slackMsgTxt = slackMsgTxt.replace(/:memo:/g, ':pencil:');
+			slackMsgTxt = slackMsgTxt.replace(/:piggy:/g, ':pig:');
+			slackMsgTxt = slackMsgTxt.replace(/:uk:/g, ':gb:');
+			slackMsgTxt = slackMsgTxt.replace(/<(http[s]?:[^>]*)>/g, '$1');
+
+			slackMsgTxt.replace(/(?:<@)([a-zA-Z0-9]+)(?:\|.+)?(?:>)/g, (match, userId) => {
+				if (!this.userTags[userId]) {
+					this.findUser(userId) || this.addUser(userId); // This adds userTags for the userId
+				}
+				const userTags = this.userTags[userId];
+				if (userTags) {
+					slackMsgTxt = slackMsgTxt.replace(userTags.slack, userTags.rocket);
+				}
+			});
+		} else {
+			slackMsgTxt = '';
+		}
+		return slackMsgTxt;
+	}
+
+}
+
```

Comment:
This seemed the most appropriate way to resolve 'require/dependency' so the two JS files could see each other